File: scratch/extract_isAr.py
import os
import re
import json

directory = r"c:\rafiq\rafiq_metrro\lib"
pattern = re.compile(r"isAr\s*\?\s*['\"](.*?)['\"]\s*:\s*['\"](.*?)['\"]")

results = []

for root, _, files in os.walk(directory):
    for file in files:
        if file.endswith(".dart"):
            filepath = os.path.join(root, file)
            with open(filepath, "r", encoding="utf-8") as f:
                content = f.read()
                matches = pattern.findall(content)
                if matches:
                    results.append({
                        "file": filepath,
                        "matches": matches
                    })

with open(r"c:\rafiq\rafiq_metrro\scratch\extracted_strings.json", "w", encoding="utf-8") as f:
    json.dump(results, f, ensure_ascii=False, indent=2)

print(f"Extracted {sum(len(r['matches']) for r in results)} strings from {len(results)} files.")

File: scratch/extract_isAr.dart
import 'dart:io';
import 'dart:convert';

void main() {
  final directory = Directory('lib');
  final pattern = RegExp(r"isAr\s*\?\s*['""](.*?)['""]\s*:\s*['""](.*?)['""]");
  
  final results = [];
  
  for (final entity in directory.listSync(recursive: true)) {
    if (entity is File && entity.path.endsWith('.dart')) {
      final content = entity.readAsStringSync();
      final matches = pattern.allMatches(content);
      if (matches.isNotEmpty) {
        final fileMatches = [];
        for (final match in matches) {
          fileMatches.add({
            'ar': match.group(1),
            'en': match.group(2)
          });
        }
        results.add({
          'file': entity.path,
          'matches': fileMatches
        });
      }
    }
  }
  
  final file = File('scratch/extracted_strings.json');
  file.createSync(recursive: true);
  file.writeAsStringSync(jsonEncode(results));
  
  int totalStrings = 0;
  for (var r in results) {
    totalStrings += (r['matches'] as List).length;
  }
  print('Extracted $totalStrings strings from ${results.length} files.');
}
